draft-ietf-quic-reliable-stream-reset.md

```diff
@@ -146,8 +146,11 @@ up to that byte offset are lost, the initiator MUST retransmit this data, as
 described in ({{Section 13.3 of RFC9000}}). Data sent beyond that byte offset
 SHOULD NOT be retransmitted.
 
-As described in {{Section 3.2 of RFC9000}}, it MAY deliver data beyond that
-offset to the application.
+As described in {{Section 3.2 of RFC9000}}, QUIC stacks MAY deliver data beyond
+that offset to the receiving application. An sending QUIC stack MAY choose a
```

Comment:
I don't understand this sentence. This seems to be up to the API contract between the sender's QUIC stack and the application. I'd consider it a very weird API if the application requests a reset with reliable size X, but the QUIC stack might sends a RESET_STREAM_AT with reliable size Y, but that's none of our business.

All that the protocols cares about is that a RESET_STREAM_AT was sent with offset Y.

Reply:
> I'd consider it a very weird API if the application requests a reset with reliable size X, but the QUIC stack might sends a RESET_STREAM_AT with reliable size Y, but that's none of our business.

FWIW, this actually helps reduce sender state. When application requests a Reliable Reset, [POC of quicly](https://github.com/h2o/quicly/pull/550/commits/2326f9572b8d01687f2af398d8b1cf016aee2a0c#diff-c63dc437bd0da3e0f0e1ac685a5bc07e769a0064ea96ab7159e08e4efeeaf474R6511-R6513) chooses `max(reliable_size, size_inflight)` as the value of Final Size. The only additional state is a flag indicating if the value has to be communicated using FIN or if the value has to be sent as "Final Size" and also "Reliable Size" using a RESET_STREAM_AT frame.